Components/Metrics/Impact/elxImpactMetric.h

```diff
@@ -0,0 +1,380 @@
+
+  /** Update the current iteration and refresh feature maps in static mode.
+   */
+  void
+  AfterEachIteration() override;
+
+  /**
+   * Do some things before each resolution:
+   * \li Set CheckNumberOfSamples setting
+   * \li Set UseNormalization setting
+   */
+  void
+  BeforeEachResolution() override;
+
+
+protected:
+  /** The constructor. */
+  ImpactMetric() = default;
+  /** The destructor. */
+  ~ImpactMetric() override = default;
+
+  unsigned long m_CurrentIteration;
+
+private:
+  elxOverrideGetSelfMacro;
+
+  std::vector<itk::ImpactModelConfiguration>
+  GenerateModelsConfiguration(unsigned int level,
+                              std::string  type,
+                              std::string  mode,
+                              unsigned int imageDimension,
+                              bool         useMixedPrecision);
+};
+
+/**
+ * \brief Parse a string into typed values using a custom delimiter.
+ *
+ * If `delimiter == '\0'`, characters are split one by one.
+ * Otherwise, the string is split using the given delimiter.
+ * Uses GetValueFromString<T>() for each token.
+ *
+ * \tparam T The type to parse.
+ * \param valueStr The input string.
+ * \param defaultValue The fallback value.
+ * \param delimiter The token delimiter or `\0` for char-splitting.
+ * \return A vector of parsed values.
+ */
+inline std::vector<bool>
+GetBooleanVectorFromString(std::string valueStr, bool defaultValue)
+{
+  std::vector<bool> values(defaultValue, valueStr.size());
+  for (char c : valueStr)
+  {
+    std::stringstream tokenStream(std::string(1, c));
+    std::string       subToken;
+    if (tokenStream >> subToken)
+    {
+      /** Cast the string to type bool. */
+      bool value = subToken == "1";
+      values.push_back(value);
+    }
+  }
+  if (values.empty())
+  {
+    values.push_back(defaultValue);
+  }
+  return values;
+} // end GetVectorFromString
+
+/**
+ * \brief Group a flattened space-separated string into string blocks per dimension sizes.
+ *
+ * This function splits a string like `"1 2 3 4 5"` into grouped strings according to
+ * the values in `dimensions`. If dimensions = {2, 3}, result is {"1 2", "3 4 5"}.
+ *
+ * \param valueStr The space-separated values.
+ * \param dimensions The sizes of each group.
+ * \return A vector of grouped space-separated strings.
+ */
+template <typename T>
+std::vector<std::vector<T>>
+GroupByDimensions(const std::vector<T> & values, const std::vector<unsigned int> & dimensions)
+{
+  std::vector<std::vector<T>> grouped;
+  size_t                      currentIndex = 0;
+  size_t                      n = values.size();
+
+  for (unsigned int dim : dimensions)
+  {
+    std::vector<T> group;
+    for (unsigned int i = 0; i < dim; ++i)
+    {
+      if (currentIndex < n)
+      {
+        group.push_back(values[currentIndex++]);
+      }
+      else if (!values.empty())
+      {
+        group.push_back(values.back()); // répète la dernière valeur si débordement
```

Comment:
Sorry, I think I can understand what you mean by "répète la dernière valeur si débordement", but I guess you forgot to translate to English  😃 (By default, we use US English spelling.)